notifier/notifier_test.go

```diff
@@ -250,6 +260,13 @@ func TestHandlerFull(t *testing.T) {
 	h.Send(alerts[2*maxBatchSize : 3*maxBatchSize]...)
 	h.Send(alerts[3*maxBatchSize : 4*maxBatchSize]...)
 
+	// This alert should be dropped due to the configuration
+	h.Send(&model.Alert{
+		Labels: model.LabelSet{
+			"alertname": "drop",
+		},
+	})
+
```

Comment:
I'd vote for extracting this into its own test. Relabeling is merely an orthogonal pre-processing step, so this  mostly makes this test case harder to understand. And it's fairly complex to begin with.

For unit tests it's okay to have the actual code in mind. So how about a test function that applies one label modification and one dropping behavior and checks what's queued afterwards. All within the limit of a single batch.

We could rename this test to `TestHandlerQueueing` or similar.